File: softphone-ios/MyLineSoftphone/Models/ChatMessage.swift
import Foundation
import GRDB

enum MessageStatus: String, Codable {
    case sending
    case sent
    case failed
    case received
}

/// Mirrors Android `ChatMessage.kt`.
struct ChatMessage: Codable, FetchableRecord, MutablePersistableRecord, Identifiable, Equatable {
    static let databaseTableName = "chat_messages"

    var id: Int64?
    var remoteNumber: String
    var body: String
    var isOutgoing: Bool
    var timestamp: Date = Date()
    var status: MessageStatus = .sent

    mutating func didInsert(_ inserted: InsertionSuccess) {
        id = inserted.rowID
    }
}

File: softphone-android/app/src/main/java/com/mylinetelecom/softphone/models/ChatMessage.kt
package com.mylinetelecom.softphone.models

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "chat_messages")
data class ChatMessage(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val remoteNumber: String,
    val body: String,
    val isOutgoing: Boolean,
    val timestamp: Long = System.currentTimeMillis(),
    val status: MessageStatus = MessageStatus.SENT,
    val messageType: String = "sms"   // "sms" or "whatsapp"
)

enum class MessageStatus {
    SENDING,
    SENT,
    FAILED,
    RECEIVED
}
